admin/js/ajax.js

```diff
@@ -23,4 +23,28 @@ jQuery(document).ready(function ($) {
 		});
 	});
 
+
+	//Copy the variable from the badges
+	$('.juvo_variable_badges').click(function (event) { 
+		
+		navigator.clipboard.writeText("");
+		
+		//Put the badge value in an input
+		$("#badgesInput").val(event.target.text) ;
+
+		 // Get the text field
+		 var copyText = $("#badgesInput");
+
+		 // Select the text field
+		 copyText.select();
+		 
+	   
+		 // Copy the text inside the text field
+		 navigator.clipboard.writeText(copyText.val());
+		 
+		 // Alert the copied text
+		 alert("You Copied the text : " + copyText.val());
+	});
```

Comment:
since the actual placeholder is also the inner html of the badge you could simply get that html on click of any badge. I would like that behaviour.  Maybe include https://clipboardjs.com/ for a better user experience.

Create a new javascript file in admin/js named placeholder.js. In this file, import clipboard.js. Make the new file an exported module with a static method called bindEvents where you add the event listeners. Within the event listener, a new object of the placeholder class should be instantiated and a method where the copy process takes place should be invoked.